### PR TITLE
TRT-2814: Eliminate payload_test_failures_14d_matview

### PR DESCRIPTION
**Pull request by mstaeble**
## Summary

- Replace the `payload_test_failures_14d_matview` materialized view with a direct SQL query that leverages partition pruning on `prow_job_run_tests`
- Fix sort.Slice comparators to use strict weak ordering with deterministic tiebreakers (FailureCount, then Name)
- Add `ORDER BY pj.name` to all `array_agg` calls for deterministic parallel array ordering
- Fix blocker score reason string from "most recent payload was Accepted" to "most recent payload was not Rejected" (handles non-Accepted/non-Rejected phases like "Ready")
- Reject sort parameters on the `/api/releases/test_failures` endpoint (400) since sorting is handled server-side
- Fix `FailedPayloads` doc comment to not say "rejected payload" since the map includes all payloads
- Add partition pruning filter to `scripts/rejected-payloads.py`
- Add 36 integration tests covering payload test failure queries, blocker scoring, streak override, sort tiebreakers, and edge cases

## Test plan

- [x] `make lint` passes clean (0 Go issues)
- [x] `make integration` passes (36 integration tests)
- [x] Deploy to staging and benchmark against current matview-based implementation
- [x] Verify API response parity with staging server

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Improved payload failure analysis across the last 14 days with stream/architecture filtering.
  * Added richer failure results, including related failed jobs and job run links.
* **Bug Fixes**
  * Updated blocker scoring explanations and overall ranking (blocker score, then failure count, then name).
  * Adjusted payload failure ordering in payload-level views to match the new ranking rules.
* **API Behavior**
  * Payload analysis endpoint now rejects requests that specify unsupported sorting.
* **Tests**
  * Added integration coverage for filtering, aggregation, scoring, and sort tie-breakers.

### CODE REVIEW

File: pkg/db/query/payload_queries.go [hunk 1]
```diff
@@ -47,6 +47,45 @@ func GetLastAcceptedByArchitectureAndStream(db *gorm.DB, release string, reportE
...
+JOIN prow_job_run_tests pjrt ON pjrt.prow_job_run_id = pjr.id
+                            AND pjrt.prow_job_run_timestamp = pjr.timestamp
+JOIN tests t ON t.id = pjrt.test_id
+JOIN prow_jobs pj ON pj.id = pjr.prow_job_id
+WHERE rt.release = ?
+  AND rt.architecture = ?
+  AND rt.stream = ?
+  AND rt.release_time >= ?
+  AND pjrt.prow_job_run_release = ?
+  AND pjrt.prow_job_run_timestamp >= ?
```
Comment (neisw) [inline pkg/db/query/payload_queries.go]: prow_job_runs should be [partitioned soonish](https://github.com/openshift/sippy/blob/main/docs/plans/trt-1989-phase4-partitioned-tables.md#phase-4b-future-phase-remaining-prowjobrun-based-tables).  It looks like it has an index for release and timestamp (idx_prow_job_runs_release_timestamp) should we just add the filtering in now so we don't have to update it later?

I don't see that the original plan included release_job_runs for partitioning

File: scripts/rejected-payloads.py [hunk 2]
```diff
@@ -24,13 +24,34 @@ class ReleaseTags(base):
...
+    WHERE
+        rt.release_tag = :tag
+        AND rjr.release_tag_id = rt.id
+        AND rjr.kind = 'Blocking'
+        AND rjr.state = 'Failed'
+        AND pjrt.prow_job_run_id = rjr.prow_job_run_id
+        AND pjrt.prow_job_run_release = :release
+        AND pjrt.prow_job_run_timestamp >= :release_time
+        AND pjrt.prow_job_run_timestamp = pjr.timestamp
+        AND pjr.timestamp >= :release_time
```
Comment (neisw) [inline scripts/rejected-payloads.py]: same prow_job_runs filter / index question here

File: pkg/db/query/payload_queries.go (hunk 1 above)
Comment (mstaeble) [reply pkg/db/query/payload_queries.go]: Yes, absolutely. Let me do that.

Comment (mstaeble) [reply pkg/db/query/payload_queries.go]: Done in [bc9d5e6](https://github.com/openshift/sippy/commit/bc9d5e62a40e89b7007f23a826c3633fecc57a91).

File: scripts/rejected-payloads.py (hunk 2 above)
Comment (mstaeble) [reply scripts/rejected-payloads.py]: Done in [bc9d5e6](https://github.com/openshift/sippy/commit/bc9d5e62a40e89b7007f23a826c3633fecc57a91).

File: pkg/api/releases.go [hunk 3]
```diff
@@ -128,34 +141,51 @@ func GetPayloadStreamTestFailures(dbc *db.DB, release, stream, arch string, filt
...
 	q, err = filter.FilterableDBResult(q, filterOpts, nil)
 	if err != nil {
 		return nil, err
 	}
-	if err := q.Find(&failedTests).Error; err != nil {
+
+	var rows []testFailureRow
+	if err := q.Scan(&rows).Error; err != nil {
 		return nil, err
 	}
```
[the remark this answers is not in the document]
Comment (mstaeble) [reply pkg/api/releases.go:155]: Rejected limit, like we do for sort field.

File: pkg/db/query/payload_queries.go [hunk 4]
```diff
@@ -62,26 +103,21 @@ func GetTestFailuresForPayload(db *gorm.DB, payloadTag, release string, releaseT
...
+	                      AND pjr.prow_job_release = ?
+	                      AND pjr.timestamp >= ?
+	JOIN prow_job_run_tests pjrt ON pjrt.prow_job_run_id = pjr.id
+	                            AND pjrt.prow_job_run_timestamp = pjr.timestamp
+	                            AND pjrt.prow_job_run_release = ?
+	                            AND pjrt.prow_job_run_timestamp >= ?
+	                            AND pjrt.status = 12
+	JOIN tests t ON t.id = pjrt.test_id
+	JOIN prow_jobs pj ON pj.id = pjr.prow_job_id
+	WHERE rt.release_tag = ?
```
Comment (neisw) [inline pkg/db/query/payload_queries.go:119]: I missed this one earlier but the previous query was filtering on pjrt timestamp and release, shouldn't we keep that and add in pjr as well?

```
	WHERE
	rt.release_tag = ?
	AND rjr.release_tag_id = rt.id
	/*AND rjr.kind = 'Blocking'*/
	AND rjr.State = 'Failed'
	AND pjrt.prow_job_run_id = rjr.prow_job_run_id
	AND pjrt.prow_job_run_release = ?
	AND pjrt.prow_job_run_timestamp >= ?
...
```

Comment (neisw) [reply pkg/db/query/payload_queries.go:119]: Or is it the joins effectively doing that

Comment (neisw) [reply pkg/db/query/payload_queries.go:119]: All good for inner joins, ignore me.
